src/tools/crossgen2/Common/TypeSystem/Common/MethodDesc.ToString.cs

```diff
@@ -49,7 +49,14 @@ public override string ToString()
             // (Skipping return type to keep things short)
             sb.Append(OwningType);
             sb.Append('.');
-            sb.Append(Name);
+            try
+            {
+                sb.Append(Name);
+            }
+            catch
+            {
+                sb.Append(DiagnosticName);
```

Comment:
Would it make sense to put methods on the same plan (DiagnostiName never throws and returns the Name or the "next best thing").

Reply:
Eh, probably. I've done it.